### PR TITLE
Add final draft of bad rows debug guide

### PR DESCRIPTION
**Pull request by colmsnowplow**

### CODE REVIEW

File: _posts/2018-12-18-debugging-bad-rows-on-gcp/index.md [hunk 1]
```diff
@@ -0,0 +1,216 @@
...
+ORDER BY 1,3 DESC
+```
+
+The `UNNEST` function is used because the errors object contains an array of `STRUCT` objects - so do note that rows with more than one error message will be counted against both messages.
+
+The output of this query looks something like this:
+
+![counts per message][counts-per-message]
+
+Some error messages will already give you an indication of what to fix - for example a schema path issue indicate that either the schema wasn't correctly uploaded to Iglu, or the tracker references the wrong path.
```
Comment (colmsnowplow) [inline _posts/2018-12-18-debugging-bad-rows-on-gcp/index.md]: Add "But for some, you'll also need to look at the custom data that created the error, to figure out exactly where the validation failure comes from"

Comment (colmsnowplow) [reply _posts/2018-12-18-debugging-bad-rows-on-gcp/index.md]: "Which we will do in the next section."

File: _posts/2018-12-18-debugging-bad-rows-on-gcp/index.md [hunk 2]
```diff
@@ -0,0 +1,216 @@
...
+
+At this point, it's a good idea to set up some monitoring dashboards for bad rows. [You can find a guide to doing this using Data Studio here][Data Studio guide]
```
Comment (colmsnowplow) [inline _posts/2018-12-18-debugging-bad-rows-on-gcp/index.md]: This is the guide that doesn't exist yet. Maybe just take this line out and we can edit it when we release that post.

Comment (Mandelliant) [reply _posts/2018-12-18-debugging-bad-rows-on-gcp/index.md]: It's all good, I replaced it with:

`At this point, it's a good idea to set up some monitoring dashboards for bad rows (a guide for doing this in Google Data Studio is forthcoming).`

It's good to tease some of the cool stuff we have coming up

File: _posts/2018-12-18-debugging-bad-rows-on-gcp/index.md [hunk 3]
```diff
@@ -0,0 +1,216 @@
...
+### External vs Native tables
+
+BigQuery allows us to query data from Cloud Storage in one of two ways:
+
+1. External tables scan the data from Cloud Storage - this has the advantage of always querying the latest data. However, while it's possible to limit the amount of data scanned, external tables don't take advantage caching, and queries can be slower.
+
+
+2. Native tables import the data into BigQuery and allow you to query from there. There are no data transfer charges from Cloud Storage but the normal charges per scanned data apply. With native tables, you can only see the data you imported when you created the table - which is a manual process. Queries will likely be faster than for external tables.
+
+In this guide we're going to use an external table for monitoring/diagnosis of the problem, and simply use a `CREATE TABLE AS` statement to generate a smaller native table and dig into the specifics. That way we create the external table once, and our volume management strategy is to simply use partitions and subset that table when we need a native one.
```
[the remark this answers is not in the document]
Comment (Mandelliant) [reply _posts/2018-12-18-debugging-bad-rows-on-gcp/index.md]: @colmsnowplow to be clear, you mean replace "use a `CREATE TABLE AS` statement" with "manually create a native table"?

The final sentence would be: `In this guide we're going to use an external table for monitoring/diagnosis of the problem, and simply manually create a native table to dig into the specifics.`

File: _posts/2018-12-18-debugging-bad-rows-on-gcp/index.md [hunk 4]
```diff
@@ -108,16 +111,16 @@ The above allows us to count bad rows over time - so we can see what errors we s
 > Unrecognized event [null]
 > Payload with vendor [] and version [] not supported by this version of Scala Common Enrich
 
-These aren't Snowplow events - they're mostly caused by requests hitting the Snwoplow collector, which didn't come from a Snowplow tracker.
+These aren't Snowplow events - they're mostly caused by requests which didn't come from a Snowplow tracker hitting the Snowplow collector.
```
Comment (colmsnowplow) [inline _posts/2018-12-18-debugging-bad-rows-on-gcp/index.md]: Let's just remove 'hitting the snowplow collector' from this sentence.

Comment (colmsnowplow) [reply _posts/2018-12-18-debugging-bad-rows-on-gcp/index.md]: It's syntactically confusing but also not necessary

Comment (Mandelliant) [reply _posts/2018-12-18-debugging-bad-rows-on-gcp/index.md]: will do

**Pull request merged**